zerver/tornado/event_queue.py

```diff
@@ -973,12 +991,33 @@ def process_message_event(
 
     presence_idle_user_ids = set(event_template.get("presence_idle_user_ids", []))
     online_push_user_ids = set(event_template.get("online_push_user_ids", []))
-    pm_mention_push_disabled_user_ids = set(
-        event_template.get("pm_mention_push_disabled_user_ids", [])
-    )
-    pm_mention_email_disabled_user_ids = set(
-        event_template.get("pm_mention_email_disabled_user_ids", [])
-    )
+
+    # TODO/compatibility: Translation code for the rename of
+    # `pm_mention_push_disabled_user_ids` to `dm_mention_push_disabled_user_ids`.
+    # Remove this when one can no longer directly upgrade from 7.x to main.
+    dm_mention_push_disabled_user_ids = set()
+    if "dm_mention_push_disabled_user_ids" in event_template:
+        dm_mention_push_disabled_user_ids = set(
+            event_template.get("dm_mention_push_disabled_user_ids", [])
+        )
+    elif "pm_mention_push_disabled_user_ids" in event_template:
+        dm_mention_push_disabled_user_ids = set(
+            event_template.get("pm_mention_push_disabled_user_ids", [])
+        )
+
+    # TODO/compatibility: Translation code for the rename of
+    # `pm_mention_email_disabled_user_ids` to `dm_mention_email_disabled_user_ids`.
+    # Remove this when one can no longer directly upgrade from 7.x to main.
+    dm_mention_email_disabled_user_ids = set()
+    if "dm_mention_email_disabled_user_ids" in event_template:
+        dm_mention_email_disabled_user_ids = set(
+            event_template.get("dm_mention_email_disabled_user_ids", [])
+        )
+    elif "pm_mention_email_disabled_user_ids" in event_template:
+        dm_mention_email_disabled_user_ids = set(
+            event_template.get("pm_mention_email_disabled_user_ids", [])
```

Comment:
If we are doing the `x in y` check earlier, we don't need to provide a default value of `[]` when we do `.get(x)`.

Reply:
Yeah, this would be a nice simplification.

Reply:
Prakhar mentioned this is PM, the issue here is the `set` function cannot handle an `Optional` argument (which `.get` returns without a provided default)

> zerver/tornado/event_queue.py:1001: error: Argument 1 to "set" has incompatible type "Optional[Any]"; expected "Iterable[Any]" [arg-type]

Reply:
OK.